src/dndbeyond/base/utils.js

```diff
@@ -283,19 +283,29 @@ async function buildAttackRoll(character, attack_source, name, description, prop
                     if (rule == CriticalRules.HOMEBREW_MAX) {
                         crit_damages.push(damagesToCrits(character, [brutal_dmg])[0]);
                     } else {
-                        // Apply great weapon fighting to brutal damage dice
-                        // TODO need to support 2024 version for GWF and Polearm Master (action name is pole strike now) and feats can be mixed
-                        if (character.hasGreatWeaponFighting(2014) &&
-                            ((properties["Attack Type"] == "Melee" &&
-                            ((properties["Properties"].includes("Versatile") && character.getSetting("versatile-choice") != "one") || properties["Properties"].includes("Two-Handed"))) ||
-                            name == "Polearm Master - Bonus Attack")) {
-                            brutal_dmg += "ro<=2"
+                        // check for both versions
+                        if (name.includes("Polearm Master") && (character.hasFeat("Polearm Master", false)) || properties["Attack Type"] == "Melee" && (properties["Properties"].includes("Versatile") || properties["Properties"].includes("Two-Handed"))) {
```

Comment:
The statement is confusing, you basically have it like this:
```js
if (n && (p) || m && (v || t))
```
The `p`, or originally `(character.hasFeat("Polearm Master", false))` is between parenthesis on its own which confused me in trying to figure out the logic... (I thought "(hasPolearm || MelleeAttack)" which made no sense).
Then you also have a mix of `&&` and `||` that aren't grouped in parenthesis, so I can't quite figure out the order of operations here, or how to read it. I think it's likely to cause the wrong logic to execute.